### ISSUE TITLE
2 USB endpoints for HID

### ISSUE THREAD

**Issue by gfwilliams** [espruino/Espruino#562]
At some point we'll allow the full USB descriptor to be fully changed, which will sort out the HID problems on Windows (which doesn't seem to like enumerating the current composite HID+CDC devices).

However it's [been suggested](https://github.com/espruino/Espruino/issues/221#issuecomment-131428582) that it would be good to be able to support two HID devices (eg. mouse + keyboard), so it would be worth trying to arrange that two endpoints could be made available (maybe with a modification to the `E.sendUSBHID` command

**Comment by t3rminus** [comment]
This sounds great! I know HID is pretty new, but I bought the pico to develop some custom HID stuff. Don't care as much about 2 endpoints, but being able to change the descriptor and receive data would be amazing.

**Comment by gfwilliams** [comment]
If it's just HID you care about then you can do custom descriptors at least - like is done for the USBKeyboard/etc modules: https://github.com/espruino/EspruinoDocs/blob/master/modules/USBKeyboard.js

But yes, we'd have to do wait for this issue to get solved before you can receive

**Comment by t3rminus** [comment]
Yep! Been doing that. I meant it would be nice to be able to change the Product/Manufacturer strings, as well.

**Comment by bmatusiak** [comment]
i know this is old..  i see there is  E.sendUSBHID(data)  . now we need E.readUSBHID

**Comment by gfwilliams** [comment]
For Bluetooth devices we have an `NRF.HID` event (http://www.espruino.com/Reference#l_NRF_HID) so something similar could be done with USB I guess.

However I'm afraid it's pretty low priority at the moment - which device were you looking at using this with?

**Comment by bmatusiak** [comment]
pinless pico usb

**Comment by gfwilliams** [comment]
Ok, thanks! Just to say the new WiFi almost certainly won't use STM32 (it'll be Nordic nRF52840). I'm not sure what the situation will be with USB HID support, but it'll definitely support 2 way HID via BLE out of the box
